src/snmalloc/backend_helpers/staticconditionalrange.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+#include "../pal/pal.h"
+#include "empty_range.h"
+#include "range_helpers.h"
+
+namespace snmalloc
+{
+  template<typename OptionalRange>
+  struct StaticConditionalRange
+  {
+    // This is a range that can bypass the OptionalRange if it is disabled.
+    // Disabling is global, and not local.
+    // This is used to allow disabling thread local buddy allocators when the
+    // initial fixed size heap is small.
+    // 
+    // The range builds a more complex parent 
+    //    Pipe<ParentRange, OptionalRange>
+    // and uses the ancestor functions to bypass the OptionalRange if the flag
+    // has been set.
+    template<typename ParentRange>
+    class Type : public ContainsParent<Pipe<ParentRange, OptionalRange>>
```

Comment:
I've nothing to add here except to quote Hackers:

    It's so neat and clean.
    Doesn't look like a hacker wrote it.

Reply:
Now I need to watch "Hackers" ;)